cmd/simv3/simulator/upkeep/pipeline.go

```diff
@@ -0,0 +1,188 @@
+package upkeep
+
+import (
+	"context"
+	"log"
+	"math/big"
+	"sync"
+
+	"github.com/smartcontractkit/ocr2keepers/cmd/simv3/config"
+	"github.com/smartcontractkit/ocr2keepers/cmd/simv3/simulator/net"
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg/v3/types"
+)
+
+type CheckTelemetry interface {
+	CheckID(string, uint64, [32]byte)
+}
+
+type CheckPipeline struct {
+	// provided dependencies
+	active         *ActiveTracker
+	performs       *PerformTracker
+	rpc            *net.SimulatedNetworkService
+	checkTelemetry CheckTelemetry
+	logger         *log.Logger
+}
+
+// TODO: provide upkeep configurations to this component
+// NewCheckPipeline ...
+func NewCheckPipeline(
+	conf config.RunBook,
+	active *ActiveTracker,
+	performs *PerformTracker,
+	netTelemetry net.NetTelemetry,
+	conTelemetry CheckTelemetry,
+	logger *log.Logger,
+) *CheckPipeline {
+	service := net.NewSimulatedNetworkService(
+		conf.RPCDetail.ErrorRate,
+		conf.RPCDetail.RateLimitThreshold,
+		conf.RPCDetail.AverageLatency,
+		netTelemetry,
+	)
+
+	return &CheckPipeline{
+		active:         active,
+		performs:       performs,
+		rpc:            service,
+		checkTelemetry: conTelemetry,
+		logger:         log.New(logger.Writer(), "[check-pipeline] ", log.Ldate|log.Ltime|log.Lshortfile),
+	}
+}
+
+// TODO: finish retry/error conditions in check pipeline
```

Comment:
Yes. I didn't want to get into it in this PR since it was already giant to start with. There are a few other TODOs to address details of the simulation.